### ISSUE TITLE
Pipeline image labels are wasteful and misleading

### ISSUE THREAD

**Issue by hannes-ucsc** [BD2KGenomics/cgl-docker-lib#191]
Here is the table that maps API version to docker version:

https://docs.docker.com/engine/reference/api/docker_remote_api/

As you can see, we don't need separate 1.9.0 and 1.9.1 images, since both versions use the same API versions. OTOH, we don't want to ask our users to figure out what API version their Docker version uses, so we'll label them with the Docker version.

We should also add a `docker` signifier to the version label.

TL;DR: Label the images `docker-1.9.x`, `docker-1.10.x` etc. and within the image install the latest `x` hotfix release you can find.

Later, we'll worry about future major Docker releases that may not increment the API version. We can always alias the images, to have two labels refer to the same image.
